src/sagemaker/local/data.py

```diff
@@ -233,8 +236,29 @@ def split(self, file):
         Returns: generator for the individual records that were split from
         the file
         """
-        with open(file, "r") as f:
-            yield f.read()
+        with open(file, "rb") as f:
+            buf = f.read()
+            if not self._is_binary(buf):
+                buf = buf.decode()
+            yield buf
+
+    def _is_binary(self, buf):
+        """binary check.
+
+        binary or text check.
+
+        binary charactor is below.(non-utf8)
+        0x07-0x0a, 0x0c-0x0d, 0x20-0x7e, 0x80-0xFF
+
+        if there are include, it's binary judgment.
+
+        Args:
+            buf (bytes): bytes in target file.
+
+        Returns:
+            is binary(True) or text(False).
+        """
+        return bool(buf.translate(None, self._textchars))
```

Comment:
this isn't a correct implementation. This will return True whether the buf array contains text or non-text bytes, as long as buf still contains *anything* after the deletions have been processed.

Reply:
I'm sorry, I couldn't understand.


This case is _is_binary = False

```
    test_file_path = tmpdir.join("none_test.txt")

    with test_file_path.open("w") as f:
        f.write("this\nis\na\ntest")

    data = [x for x in splitter.split(str(test_file_path))]
    assert data == ["this\nis\na\ntest"]

```


This case is _is_binary = True

```
    test_bin_file_path = tmpdir.join("none_test.bin")

    with test_bin_file_path.open("wb") as f:
        f.write(b"\xff\xd8\xff\xe0\x00\x10JFIF\x00\x01\x01\x00\x00\x01\x00\x01\x00\x00C")

    data = [x for x in splitter.split(str(test_bin_file_path))]
    assert data == [b"\xff\xd8\xff\xe0\x00\x10JFIF\x00\x01\x01\x00\x00\x01\x00\x01\x00\x00C"]

```

Is this test insufficient?

Reply:
yes, the test looks code. i did some additional testing with non-latin text and it looks like it is working correctly.